YuliiaShaposhnikova/RPSLS_Game/rpsls_tk_game.py

```diff
@@ -0,0 +1,149 @@
+import random
```

Comment:
here is more clear code 

import random
from tkinter import Tk, Label, Button, messagebox

def center_window(window):
    window.update_idletasks()
    width = window.winfo_width()
    height = window.winfo_height()
    screen_width = window.winfo_screenwidth()
    screen_height = window.winfo_screenheight()
    x = (screen_width // 2) - (width // 2)
    y = (screen_height // 2) - (height // 2)
    window.geometry(f"{width}x{height}+{x}+{y}")


def play_game(player_choice):
    sheldons_fighter = random.choice(['Rock', 'Paper', 'Scissors', 'Lizard', 'Spock'])
    sheldons_fighter_label.config(text=f"My fighter is {sheldons_fighter}")

    results = {
        'Rock': {
            'Rock': "Dead head!",
            'Paper': 'My paper covers your rock! You lose!',
            'Scissors': 'Your rock crushes my scissors! You win!',
            'Lizard': 'Your rock crushes my lizard! You win!',
            'Spock': 'My Spock vaporizes your rock! You lose!'
        },
        'Paper': {
            'Rock': "Dead head!",
            'Paper': 'Dead head!',
            'Scissors': 'My scissors cuts your paper! You lose!',
            'Lizard': 'My lizard eats your paper! You lose!',
            'Spock': 'Your paper disproves my Spock! You win!'
        },
        'Scissors': {
            'Rock': 'My rock crushes your scissors! You lose',
            'Paper': 'Your scissors cuts my paper! You win!',
            'Scissors': "Dead head!",
            'Lizard': 'Your scissors decapitates my lizard! You win!',
            'Spock': 'My Spock smashes your scissors! You lose!'
        },
        'Lizard': {
            'Rock': 'My rock crushes your lizard! You lose!',
            'Paper': 'Your lizard eats my paper! You win!',
            'Scissors': 'Your scissors decapitates my lizard! You lose!',
            'Lizard': "Dead head!",
            'Spock': 'Your lizard poisons my Spock! You win!'
        },
        'Spock': {
            'Rock': 'Your Spock vaporizes my rock! You win!',
            'Paper': 'My paper disproves your Spock! You lose!',
            'Scissors': 'Your Spock smashes my scissors! You win!',
            'Lizard': 'My lizard poisons your Spock! You lose!',
            'Spock': "Dead head!"
        }
    }

    result = results[player_choice][sheldons_fighter]
    result_label.config(text=result)
    root.after(1000, play_again)


def play_again():
    answer = messagebox.askyesno("Play Again", "Wanna play again?")
    if answer:
        result_label.config(text="")
        sheldons_fighter_label.config(text="")
        title_label1.config(text="")
        root.withdraw()
        root.deiconify()
    else:
        root.quit()


root = Tk()
root.geometry('430x300')
root.resizable(False, False)
root.title('Rock Paper Scissors Lizard Spock')

title_label1 = Label(root, text="Hello! I'm Sheldon Cooper and I want you to play with me", font=("Arial", 16))
title_label1.grid(row=0, column=0, columnspan=5, padx=5, pady=10)

title_label2 = Label(root, text="Choose your fighter", font=("Arial", 15))
title_label2.grid(row=1, column=0, columnspan=5, padx=5, pady=10)

sheldons_fighter_label = Label(root, text="", font=("Arial", 14))
sheldons_fighter_label.grid(row=4, column=0, columnspan=5, padx=5, pady=5)

result_label = Label(root, text="", font=("Arial", 16))
result_label.grid(row=5, column=0, columnspan=5, padx=5, pady=10)

button1 = Button(root, text="Rock", command=lambda: play_game('Rock'))
button1.grid(row=2, column=0, padx=5, pady=5)

button2 = Button(root, text="Paper", command=lambda: play_game('Paper'))
button2.grid(row=2, column=1, padx=5, pady=5)

button3 = Button(root, text="Scissors", command=lambda: play_game('Scissors'))
button3.grid(row=2, column=2, padx=5, pady=5)

button4 = Button(root, text="Lizard", command=lambda: play_game('Lizard'))
button4.grid(row=2, column=3, padx=5, pady=5)

button5 = Button(root, text="Spock", command=lambda: play_game('Spock'))
button5.grid(row=2, column=4, padx=5, pady=5)

center_window(root)
root.mainloop()